infra/templates/az-isolated-service-single-region/query_subnet_vnet_ids.sh

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# Check if `jq` is installed. If not, fail fast.
+command -v jq >/dev/null 2>&1 || { echo >&2 "jq is not installed. Please re-run after installing"; exit 1; }
+
+# This command queries for the subnets in a VNET
+# and emits them in the following format:
+#   [
+#       { name: <subnet name 1>, id: <subnet ID 1> },
+#       { name: <subnet name 2>, id: <subnet ID 2> },
+#       ...
+#   ]
+SUBNET_LISTING=$(az network vnet subnet list \
+    --subscription $1       \
+    --resource-group $2     \
+    --vnet-name $3          \
+    --output json           \
```

Comment:
Unless you need the JSON, you can use `--output tsv` which is tab-separated values, with no keys or simply "plain text". Should remove the dependency on jq...

Reply:
Unfortunately terraform only accepts JSON from external programs. Even more of a pain is that it only accepts string keys and string values. This is why I am using `jq` to transform the json output.

I tried really hard to avoid this but was unable to. Happy for an alternative solution, though! LMK if anything comes to mind.